### DISCUSSION TITLE
Console Collapse?

### DISCUSSION THREAD

**Discussion by Brad123ghost** [ANU-CHEEKI-BREEKI/com.anupackages.debugconsole#6] [Q&A] [answered]
I have constant debug messages and either it can't scroll as fast or is a bit delayed. Is there a collapse feature like the Unity one where it only shows new messages once? Is that a possible implementation that can be achieved with the console?

**Comment by ANU-CHEEKI-BREEKI** [comment, marked as answer]
Hi, thanks for the question!

Unfortunately, collapsing messages is not implemented yet 😔 
And i will add this to my TODO list right now!

However, as a quick workaround you can modity `Add` method in `ANU.IngameDebug.Console.LogsContainer`
like this, to force all messages be collapsed by default:
````
public void Add(Log log)
{
    log.WholeIndex = _allLogs.Count;

    var alreadyContains = _allLogs.Any(l 
        => l.StackTrace == log.StackTrace
        && l.Message == log.Message
        && l.MessageType == log.MessageType
        && l.ConsoleLogtype == log.ConsoleLogtype
    );

    if (!alreadyContains)
    {
        _allLogs.AddLast(log);
        AddAsFiltered(log);
    }

    var types = (log.ConsoleLogtype, log.MessageType);
    if (!_messagesCount.ContainsKey(types))
        _messagesCount[types] = 0;
    _messagesCount[types]++;

    Added?.Invoke(new CollectionChangedArgs(log, this));
}
````

Its not gonna be too performance when you already have many different logs, but in general, as a quick fix, it should work ok.

**Reply by Brad123ghost** [reply]
Thanks, this works perfect in my case :)
